### DISCUSSION TITLE
How to change size of grid

### DISCUSSION THREAD

**Discussion by mcp292** [vega/altair#3153] [Q&A]
# [MWE](https://altair-viz.github.io/gallery/simple_line_chart.html)

```py
import altair as alt
import numpy as np
import pandas as pd


x = np.arange(100)
source = pd.DataFrame({
  'x': x,
  'f(x)': np.sin(x / 5)
})

alt.Chart(source).mark_line().encode(
    x=alt.X('x').axis(tickMinStep=1),
    y=alt.Y('f(x)').axis(tickMinStep=1)
).save("grid_step.png")

alt.Chart(source).mark_line().encode(
    x=alt.X('x').axis(tickCount=100),
    y=alt.Y('f(x)').axis(tickCount=100)
).save("grid_count.png")
```

# Problem 

I would like to make a grid line appear at every single unit.

When using `tickMinStep=1` nothing happens:
![step](https://github.com/altair-viz/altair/assets/49123398/45afb8ec-60e7-45c4-8f1f-108865f69ace)

But [the documentation](https://altair-viz.github.io/user_guide/generated/core/altair.Axis.html) makes it seem this is exactly what I want:

> The minimum desired step between axis ticks, in terms of scale domain values. For example, a value of 1 indicates that ticks should not be less than 1 unit apart. If tickMinStep is specified, the tickCount value will be adjusted, if necessary, to enforce the minimum step value.

Using `tickCount` seems to work, but requires that you know the bounds of your data a priori:
![count](https://github.com/altair-viz/altair/assets/49123398/89630d84-b1c3-4a26-a0f8-c3db6b96d984)

# Notes

I would prefer to use `tickMinStep` or similar because I would like to specify the interval/spacing of the gridlines, without knowing the domain min and max.

This answer might be related, but I don't know how to translate it: https://github.com/vega/vega/issues/583#issuecomment-322338211.

**Comment by ChristopherDavisUCI** [comment]
What I’m suggesting feels hacky, but what if you specify both a large
tickCount (like 1000) and tickMinStep=1.  Does that produce what you want?
I think you are hoping for a tickMaxStep rather than a tickMinStep, but it
doesn’t seem to exist.

**Reply by mcp292** [reply]
I agree, `tickMaxStep` is actually what I need. That does feel hacky, I wouldn't move forward with that.

**Comment by rambip** [comment]
@mcp292 I agree with you.

If you still think this is an important missing feature, maybe try to defend it over there: https://github.com/vega/vega-lite/issues/9594
